test/db/mysql/unit_test.rb

```diff
@@ -96,7 +96,7 @@ def initialize; end
       connection_handler = connection_handler_stub
 
       config = { :jndi => 'jdbc/TestDS' }
-      connection_handler.expects(:jndi_connection)
+      connection_handler.expects(:jndi_connection).with() { |c| config = c }
```

Comment:
just a minor thingy ... its morning here but did you not mean `config == c` ?

Reply:
Nope. Since config is now cloned in the connection handler, we cannot check for changes on the original object. By catching the `jndi_connection` arg I can get the cloned and changed object out this way. It's a bit ugly but it was the easiest way I could come up with

Reply:
ooh so that is what this accomplishes, thanks!